osquery/sql/virtual_table.cpp

```diff
@@ -789,22 +792,38 @@ static int xBestIndex(sqlite3_vtab* tab, sqlite3_index_info* pIdxInfo) {
           column_name = std::get<0>(columns[real_column_index]);
         }
         colsUsed.insert(column_name);
+
+        const auto& options = std::get<2>(columns[i]);
+        if (options & ColumnOptions::REQUIRED) {
+          numRequiredColumns++;
+        } else if (options &
+                   (ColumnOptions::INDEX | ColumnOptions::ADDITIONAL)) {
+          numIndexedColumns++;
+        }
       }
     }
   }
 
   pIdxInfo->idxNum = static_cast<int>(kConstraintIndexID++);
-#if defined(DEBUG)
-  plan("Recording constraint set for table: " + pVtab->content->name +
-       " [cost=" + std::to_string(cost) +
-       " size=" + std::to_string(constraints.size()) +
-       " idx=" + std::to_string(pIdxInfo->idxNum) + "]");
-#endif
+  if (FLAGS_planner) {
+    plan("xBestIndex Recording constraint set for table: " +
```

Comment:
See above about `FLAGS_planner` check and `xBestIndex` in the log line.

Reply:
I added check for FLAGS_planner to avoid doing the script concat when flag not enabled.